src/coreclr/pal/src/CMakeLists.txt

```diff
@@ -112,10 +112,6 @@ if(CLR_CMAKE_HOST_ARCH_ARM)
   endif()
 endif(CLR_CMAKE_HOST_ARCH_ARM)
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
```

Comment:
Can you please give an example why the change is needed? Is it assembler related?

Reply:
This should stay, but the condition should be updated so that it gets added only for  Intel processors (CLR_CMAKE_HOST_ARCH_AMD64 or CLR_CMAKE_HOST_ARCH_I386).

Reply:
arm assembler doesn't have this option, and linux x64 shows:
```
$ as --help|grep divide
  --divide                ignored
```

Why is this option needed?

Reply:
This was added for SunOS support. @am11 do you remember why was it needed?

Reply:
On SYS-V derivative platforms, `/` in assembly is treated as `#` rather than division. In context2.S, we are using division, so it breaks build on platforms, such as SunOS-likes and QNX.

Please note that GNU toolchain support is strictly added for community supported platforms in .NET, where LLVM toolchain is either not available or not as complete. If your platform  already has good support for LLVM toolchain, it is recommended (per its official support stauts) to use that instead.

Reply:
Thanks for explanation. I've kept the original GNU condition and added x86/x64 check.